packages/key-management/src/util/mapHardwareSigningData.ts

```diff
@@ -87,6 +87,12 @@ const matchGroupedAddress = (knownAddresses: GroupedAddress[], outputAddress: Bu
   return knownAddresses.find(({ address }) => address === outputAddressBech32);
 };
 
+export const mintValueAsString = (scope: ManagedFreeableScope, amount: CML.Int): string => {
+  const positiveAmount = scope.manage(amount?.as_positive())?.to_str();
+  const negativeAmount = scope.manage(amount?.as_negative())?.to_str();
+  return positiveAmount ? positiveAmount : `-${negativeAmount}`;
```

Comment:
In the case of `undefined | null` amount, the function will return `-undefined`

Reply:
Nice catch, although minting an `undefined` or `null` amount is not a thing, so shouldn't really be a problem

Reply:
True. In this case, we shouldn't even take it into account:
```diff
- amount?.as_positive()
+ amount.as_positive()

Reply:
> True. In this case, we shouldn't even take it into account:

Yes, the optional chaining operator is wrong. Gone in 67e3f4aa61259851b735f3652c1fd95e64000848